### PR TITLE
feat(bridge): bridge textDocument/onTypeFormatting with config-driven triggers

### LINKED ISSUE #354: Bridge textDocument/onTypeFormatting: trigger characters unknown until downstream servers spawn
## Context

Part of the effort to cover more `textDocument/*` methods (LSP 3.18). The bridging itself is mechanical (position-based request returning `TextEdit[]`, like rangeFormatting), but the capability advertisement needs discussion.

## Problem

`documentOnTypeFormattingProvider` requires a concrete `firstTriggerCharacter` (+ `moreTriggerCharacter`) in the initialize response. kakehashi spawns downstream servers lazily, so at initialize time it cannot know which trigger characters the downstream servers will declare (e.g. `}` and `;` for clangd, `\n` for many).

## Options

1. **Fixed superset**: advertise a common set (`\n`, `}`, `;`, `)`) and forward only when the downstream declares the character. Cheap, but fires spurious requests on every keystroke matching the superset.
2. **Config-driven**: let users declare trigger characters per bridge server in settings; advertise the union.
3. **Dynamic registration**: register `textDocument/onTypeFormatting` via `client/registerCapability` after the first downstream spawn reveals its trigger characters (requires client support for dynamic registration).

Option 3 is the most correct but kakehashi currently performs no upstream dynamic registration; that machinery would also benefit other capabilities (e.g. completion trigger characters are currently hardcoded to `.` and `:`).

### PR DESCRIPTION
**Pull request by atusy**
Closes #354

## Summary

Bridges `textDocument/onTypeFormatting` to downstream language servers for injected regions, using the maintainer-approved **option 2 (config-driven triggers)**:

- `documentOnTypeFormattingProvider` requires concrete trigger characters at initialize, but downstream servers spawn lazily — so users declare `onTypeFormattingTriggers` per `languageServers` entry (wildcard-mergeable, overlay-wins). kakehashi advertises the **sorted, deduplicated union**; no config keeps the capability unadvertised, so users who don't opt in see zero new requests.
- **Double filter**: a request is forwarded to a downstream server only when that server's own `documentOnTypeFormattingProvider` declares the typed character, so a misconfigured superset degrades to no-ops rather than bad edits. Dynamically-registered providers are trusted without trigger inspection (their trigger set lives in registration options the bridge doesn't parse).

## Design

The handler mirrors hover's position-based shape (`walk_layers` + per-region `dispatch_preferred` under its own `textDocument/onTypeFormatting` aggregation key); the response path reuses the formatting family's virtual→host `TextEdit` translation including past-EOF protection, and `Some(vec![])` keeps its authoritative "no edits" semantics.

## Tests

- Unit: trigger-union determinism (sorted/deduped, order-independent), overlay-wins merge, request wire shape, all four `downstream_declares_trigger` branches, `has_capability` tables.
- E2E (`tests/e2e_on_type_formatting.rs`, new `on-type` mock mode): sorted union advertised at initialize (absent without config); a declared trigger returns host-translated edits; an undeclared character is filtered bridge-side (the mock answers ANY character, so the null result can only come from the filter).

## Review notes

Pre-reviewed via 10-perspective subagent review (all findings addressed: trigger-filter unit tests, doc corrections incl. the capability-frozen-after-initialize note and the wildcard-union edge case) and a Codex session that traced the full request path and converged with no comments.

### CODE REVIEW

File: src/lsp/lsp_impl/text_document/on_type_formatting.rs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        let virt =
+            self.on_type_formatting_virt_layer(&lsp_uri, position, &params.ch, params.options);
+        self.walk_layers(
+            &lsp_uri,
+            METHOD,
+            METHOD,
+            raw_params,
+            virt,
+            parse_host_verbatim::<Vec<TextEdit>>,
+            |edits: &Vec<TextEdit>| !edits.is_empty(),
```
[the remark this answers is not in the document]
Comment (atusy) [reply src/lsp/lsp_impl/text_document/on_type_formatting.rs:38]: This predicate intentionally matches the formatting family's layer-walk semantics rather than being a drop: range_formatting.rs uses the identical `|edits| !edits.is_empty()` for walk_layers. The predicate only decides whether the NEXT layer (host) gets a chance — an authoritative `Some(vec![])` from a virt server still stops the within-layer server fan-out (dispatch_preferred's `opt.is_some()`), which is where the authoritative-empty semantics matter. Upstream, `null` and `[]` are equivalent for onTypeFormatting per LSP 3.18 (`TextEdit[] | null` — both mean "no edits to apply"), so no client-visible information is lost; letting the host layer try after an empty virt result is the same cross-layer fallthrough formatting and rangeFormatting already implement. Changing it only for onTypeFormatting would make the family inconsistent — if the layer-walk-on-empty semantics should change, that's a cross-method design question better tracked separately.

File: src/lsp/bridge/text_document/on_type_formatting.rs [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+            .get_or_create_connection(server_name, server_config)
+            .await?;
+        if !handle.has_capability("textDocument/onTypeFormatting") {
+            return Ok(None);
+        }
+        let static_provider = handle
+            .server_capabilities()
+            .and_then(|caps| caps.document_on_type_formatting_provider.as_ref());
+        if !downstream_declares_trigger(static_provider, ch) {
+            log::debug!(
```
[the remark this answers is not in the document]
Comment (atusy) [reply src/lsp/bridge/text_document/on_type_formatting.rs]: Fixed in the latest commit: whenever a dynamic registration exists for textDocument/onTypeFormatting, the static trigger filter is bypassed entirely (static_provider forced to None → the existing dynamic-trust path), so dynamically-registered triggers can no longer be dropped by a stale/differing static set.

File: src/config/settings.rs [hunk 3]
```diff
@@ -272,6 +272,48 @@ pub struct BridgeServerConfig {
...
+    /// (servers spawn lazily), so users declare them here. The union across
+    /// all servers is advertised as `documentOnTypeFormattingProvider`; a
+    /// request is forwarded to a downstream server only when that server's
+    /// own capabilities also declare the typed character. Unset or empty
+    /// everywhere → the capability is not advertised (today's behavior).
+    ///
+    /// Note: the capability is negotiated once at `initialize` and cannot
+    /// change at runtime. Adding triggers via `didChangeConfiguration` affects
+    /// forwarding but does not re-advertise the capability — a restart is
+    /// required for the editor to start sending the requests.
```
[the remark this answers is not in the document]
Comment (atusy) [reply src/config/settings.rs]: Good catch — corrected. You're right that this field feeds *only* the `initialize`-time advertisement (`on_type_formatting_trigger_union` → `lifecycle.rs`); the forwarding filter (`downstream_declares_trigger`) keys off the downstream server's own `documentOnTypeFormattingProvider`, never this config field. Reworded the note to state it has no runtime effect at all — neither re-advertising the capability nor changing forwarding — so a restart is required for new triggers to take effect.
